### PR TITLE
TINY-11159: focus image dialog after closing an error alert

### PR DESCRIPTION
**Pull request by ztomaszyk**
Related Ticket: TINY-11159

Description of Changes:
* Placeholder text

Pre-checks:
* [x] Changelog entry added
* [x] Tests have been added (if applicable)
* [x] Branch prefixed with `feature/`, `hotfix/` or `spike/`

Review:
* [x] Milestone set
* [x] ~~Docs ticket created (if applicable)~~

GitHub issues (if applicable):

### CODE REVIEW

File: modules/tinymce/src/themes/silver/main/ts/ui/window/SilverDialogInstanceApi.ts [hunk 1]
```diff
@@ -152,6 +156,7 @@ const getDialogApi = <T extends Dialog.DialogData>(
     setData,
     setEnabled,
     focus,
+    focusSelf,
```
Comment (TheSpyder) [inline modules/tinymce/src/themes/silver/main/ts/ui/window/SilverDialogInstanceApi.ts:159]: ooh, hang on. This is adding a public API, which has implications including documentation.

@spocke should we discuss this?

**Review by TheSpyder — CHANGES_REQUESTED** [review]
This is a nice idea, but I think it crosses the public API boundary too much.

I would instead treat it as a bug; if there is an open modal dialog the `editor.windowManager.alert()` API should return focus to it automatically.

**Comment by spocke** [comment]
> This is a nice idea, but I think it crosses the public API boundary too much.
> 
> I would instead treat it as a bug; if there is an open modal dialog the `editor.windowManager.alert()` API should return focus to it automatically.

I'm already on it we have a new approach where we will make the dropzone focusable like the other fields. We could put this back to draft in the meantime. I also don't want us to add a new public api if we don't need to.

**Comment by ztomaszyk** [comment]
The dropzone is now focusable. I marked it as ready for review.

**Review by lorenzo-pomili — APPROVED** [review]

**Review by spocke — APPROVED** [review]

File: modules/tinymce/src/plugins/image/test/ts/browser/UploadTabTest.ts [hunk 2]
```diff
@@ -203,4 +203,21 @@ describe('browser.tinymce.plugins.image.UploadTabTest', () => {
     await pAssertSrcTextValue('logo.jfif');
     closeDialog(editor);
   });
+
+  it('TINY-: DialogFocusAfterErrorTest', async () => {
```
[the remark this answers is not in the document]
Comment (ztomaszyk) [reply modules/tinymce/src/plugins/image/test/ts/browser/UploadTabTest.ts]: It was just a placeholder, forgot to change that 🤦‍♀️ Fixed!

**Pull request merged**